processor/src/main/resources/org/mapstruct/ap/internal/model/MethodReference.ftl

```diff
@@ -44,6 +44,8 @@
             <#if ext.targetBeanName??>${ext.targetBeanName}<#else>${param.variableName}</#if><#if ext.targetReadAccessorName??>.${ext.targetReadAccessorName}</#if><#t>
         <#elseif param.mappingContext>
             ${param.variableName}<#t>
+        <#elseif param.targetPropertyName>
+            "${param.targetPropertyNameValue}"<#t>
```

Comment:
> can't we retrieve this value inside the freemarker tree itself?
I can't find a way properly reference `PropertyMapping` model and its `name` property in `arguments` ftl macro.
#noob

Reply:
The `PropertyMapping` knows the `name`, so in the `PropertyMapping.ftl` you can pass it on at the `<@includeModel object=assignment` call.
You see there for example `targetType=targetType`, the first `targetType` is the name it will be known in the assignment area. There it is refered to as `ext.targetType`. The second `targetType` is the java bean `targetType` from the `PropertyMapping` class, the `getTargetType()` method.
You will need to pass it on like this to get it to where you use it at `MethodReference.ftl`.

Reply:
I'm some what guessing, but is `PropertyMapping` assignment accessible in ftl for presence check methods?
If I use:
```
@Condition
default boolean isNotBlank(String value, @TargetType Class<?> cls) {...
```
I get null for cls and NPE in `org.mapstruct.ap.internal.model.MethodReference#inferTypeWhenEnum`.

Reply:
It seems you have found a bug.

I can reproduce it with 1.5.1.Final. The current value of `@TargetType` here should have been `Class<String>` I think, which is the `TargetType` of the parameter value for which the `isNotBlank` check takes place. So don't think that `@TargetType` is that useful for `@Condition` since you are already using the correct type in the previous parameter.

Let's fix this in a separate issue (#2882) with it's own PR.

Reply:
The PR #2883 is a [nice example of how to pass on information through freemarker](https://github.com/mapstruct/mapstruct/pull/2883/files).

Reply:
What name do you suggest for template var? I think that `name` might be too generic, but on the other hand that is a field name in `PropertyMapping`?

Reply:
Ok, I think I'm done. I have one more observation regarding #2883 (I don't know if I'm right since I didn't write a test):
I believe `<#macro handleLocalVarNullCheck needs_explicit_local_var>` in CommonMacros.ftl should be fixed in the same way as `<#macro handleSourceReferenceNullCheck>` or the `Class<?> targetType` parameter passing might fail if presence check method is in `@Context` parameter.

Reply:
And another thing: I didn't pass `ext.targetPropertyName` in every possible assigment (as is `ext.targetType`) since it's only relevant in presence check methods. I hope that's ok.